api/client.go

```diff
@@ -244,11 +250,15 @@ func (c *Config) ConfigureTLS(t *TLSConfig) error {
 			return err
 		}
 		foundClientCert = true
```

Comment:
Do we need to grab the modifyLock?

Reply:
Yes, I think this is needed indeed. Since `ConfigureTLS` is already called with the lock held in `ReadEnvironment` I made a private function `configureTLS` that can be called when the lock is already held. `ConfigureTLS` now just takes the lock and calls it.